src/vivarium_nih_us_cvd/components/disease.py

```diff
@@ -85,60 +81,117 @@ def IschemicHeartDiseaseAndHeartFailure():
-        )
-    }
-    residual_heart_failure_incidence_data_funcs = {
-        "incidence_rate": lambda _, builder: builder.data.load(
+        hf_residual_incidence = builder.data.load(
             data_keys.IHD_AND_HF.INCIDENCE_HF_RESIDUAL
-        )
-    }
+        ).set_index(ARTIFACT_INDEX_COLUMNS)
+        return (
+            acute_mi_incidence
+            / (acute_mi_incidence + hf_ihd_incidence + hf_residual_incidence)
+        ).reset_index()
 
-    # transitions out of suspectible state
-    susceptible.allow_self_transitions()
-    susceptible.add_transition(
+    transient_susceptible_state.add_transition(
         acute_myocardial_infarction,
-        source_data_type="rate",
-        get_data_functions=acute_mi_incidence_data_funcs,
+        source_data_type="proportion",
+        get_data_functions={"proportion": acute_mi_after_susceptible_proportion},
     )
-    susceptible.add_transition(
+
+    def hf_ihd_after_susceptible_proportion(_, builder: Builder) -> pd.DataFrame:
+        acute_mi_incidence = builder.data.load(
+            data_keys.IHD_AND_HF.INCIDENCE_ACUTE_MI
+        ).set_index(ARTIFACT_INDEX_COLUMNS)
+        hf_ihd_incidence = builder.data.load(data_keys.IHD_AND_HF.INCIDENCE_HF_IHD).set_index(
+            ARTIFACT_INDEX_COLUMNS
+        )
+        hf_residual_incidence = builder.data.load(
+            data_keys.IHD_AND_HF.INCIDENCE_HF_RESIDUAL
+        ).set_index(ARTIFACT_INDEX_COLUMNS)
+        return (
+            hf_ihd_incidence / (acute_mi_incidence + hf_ihd_incidence + hf_residual_incidence)
+        ).reset_index()
+
+    transient_susceptible_state.add_transition(
         heart_failure_from_ihd,
-        source_data_type="rate",
-        get_data_functions=heart_failure_from_ihd_incidence_data_funcs,
+        source_data_type="proportion",
+        get_data_functions={"proportion": hf_ihd_after_susceptible_proportion},
     )
-    susceptible.add_transition(
+
+    def hf_resiudal_after_susceptible_proportion(_, builder: Builder) -> pd.DataFrame:
+        acute_mi_incidence = builder.data.load(
+            data_keys.IHD_AND_HF.INCIDENCE_ACUTE_MI
+        ).set_index(ARTIFACT_INDEX_COLUMNS)
+        hf_ihd_incidence = builder.data.load(data_keys.IHD_AND_HF.INCIDENCE_HF_IHD).set_index(
+            ARTIFACT_INDEX_COLUMNS
+        )
+        hf_residual_incidence = builder.data.load(
+            data_keys.IHD_AND_HF.INCIDENCE_HF_RESIDUAL
+        ).set_index(ARTIFACT_INDEX_COLUMNS)
+        return (
+            hf_residual_incidence
+            / (acute_mi_incidence + hf_ihd_incidence + hf_residual_incidence)
+        ).reset_index()
+
+    transient_susceptible_state.add_transition(
         residual_heart_failure,
-        source_data_type="rate",
-        get_data_functions=residual_heart_failure_incidence_data_funcs,
+        source_data_type="proportion",
+        get_data_functions={"proportion": hf_resiudal_after_susceptible_proportion},
```

Comment:
typo

Reply:
ah that explains why only copy and pasting worked! good catch